### DISCUSSION TITLE
limited access to website

### DISCUSSION THREAD

**Discussion by Fiyuva** [nuxt/nuxt#24496] [Questions]
I have a site I created using Nuxt3 and Nodejs. I deployed my site with Netlify. Only allowed IPs should enter my site because this is an admin panel and not everyone should be able to access it. How do I solve this? I would be happy if you help.

**Comment by wrkmartech** [comment]
You can try [server middleware](https://nuxt.com/docs/guide/directory-structure/server#server-middleware) and [getRequestIP](https://www.jsdocs.io/package/h3#getRequestIP) of `h3`.

For example (`server/middleware/access.ts`):
```ts
export default defineEventHandler((event) => {
  const whiteIPList = ['1.1.1.1.', '9.9.9.9.'];
  const requestIP = getRequestIP(event, { xForwardedFor: true });
  if (!whiteIPList.includes(requestIP)) {
    throw createError({
      statusCode: 403,
      message: 'Access is denied',
    });
  }
});
```

**Comment by Fiyuva** [comment]
It shouldn't even appear on Google. The existence of the site should not be obvious when someone without IP permission tries to enter the site. There are systems that do this, but I couldn't do it.

**Comment by wrkmartech** [comment]
@Fiyuva I think you should read docs of your hosting platform in this case
[for example](https://docs.netlify.com/security/secure-access-to-sites/traffic-rules/#configure-default-traffic-rules-for-sites-in-your-team)

**Reply by Fiyuva** [reply]
Thank you for your answer. I will try it.
